### PR TITLE
Taskscheduler enhancement

### PR DESCRIPTION
**Pull request by kannang83**
1. Provided the support for both local and hdfs for data locality task scheduling.
2. Wrote an example to test the data locality task scheduling for streaming and batch tasks.
3. Tested the task indexes for streaming task schedulers
4. Introduced the task graph constraints in the task graph, the user may either give the constraints or not. We have tested both the cases for streaming schedulers.

### CODE REVIEW

File: twister2/api/src/java/edu/iu/dsc/tws/api/dataobjects/DataObjectSource.java [hunk 1]
```diff
@@ -83,16 +83,14 @@ public void setEdgeName(String edgeName) {
    */
   @Override
   public void execute() {
-    LOG.fine("Context Task Index:" + context.taskIndex() + "\t" + getEdgeName());
+    LOG.info("Context Task Index:" + context.taskIndex() + "\t" + getEdgeName());
```
Comment (supunkamburugamuve) [inline twister2/api/src/java/edu/iu/dsc/tws/api/dataobjects/DataObjectSource.java]: Lets remove this log

File: twister2/data/src/main/java/edu/iu/dsc/tws/data/utils/HdfsDataContext.java [hunk 2]
```diff
@@ -34,10 +35,10 @@
       = "$HADOOP_HOME/etc/hadoop/core-site.xml";
 
   private static final String HDFS_DATA_DIRECTORY = "twister2.hdfs.data.directory";
-  private static final String HDFS_DATA_DIRECTORY_DEFAULT = "/user/";
+  private static final String HDFS_DATA_DIRECTORY_DEFAULT = "/user/kannan";
```
Comment (supunkamburugamuve) [inline twister2/data/src/main/java/edu/iu/dsc/tws/data/utils/HdfsDataContext.java]: Remove Kanna from here

File: twister2/task/src/main/java/edu/iu/dsc/tws/task/api/schedule/TaskInstanceId.java [hunk 3]
```diff
@@ -58,13 +46,26 @@ public boolean equals(Object o) {
 
     TaskInstanceId that = (TaskInstanceId) o;
 
-    return taskId == that.taskId && taskIndex == that.taskIndex
-        && (taskName != null ? taskName.equals(that.taskName) : that.taskName == null);
+//    return taskId == that.taskId
+//        && taskIndex == that.taskIndex
+//        && taskName.equals(that.taskName);
+
```
Comment (supunkamburugamuve) [inline twister2/task/src/main/java/edu/iu/dsc/tws/task/api/schedule/TaskInstanceId.java]: Remove these comments

File: twister2/task/src/main/java/edu/iu/dsc/tws/task/graph/DataFlowTaskGraph.java [hunk 4]
```diff
@@ -56,11 +62,26 @@ public void build() {
   public boolean addTaskVertex(String name, Vertex taskVertex) {
     if (!validateTaskVertex(name)) {
       addTaskVertex(taskVertex);
+      if (taskVertex.getGraphConstraintsMap() != null) {
+        //addGraphConstraints(name, taskVertex.getGraphConstraintsMap());
```
Comment (supunkamburugamuve) [inline twister2/task/src/main/java/edu/iu/dsc/tws/task/graph/DataFlowTaskGraph.java]: We cannot have an empty if satement

File: twister2/api/src/java/edu/iu/dsc/tws/api/dataobjects/DataObjectSource.java (hunk 1 above)
Comment (kannang83) [reply twister2/api/src/java/edu/iu/dsc/tws/api/dataobjects/DataObjectSource.java]: Removed this log.

File: twister2/data/src/main/java/edu/iu/dsc/tws/data/utils/HdfsDataContext.java (hunk 2 above)
Comment (kannang83) [reply twister2/data/src/main/java/edu/iu/dsc/tws/data/utils/HdfsDataContext.java]: Removed the name.

File: twister2/task/src/main/java/edu/iu/dsc/tws/task/api/schedule/TaskInstanceId.java (hunk 3 above)
Comment (kannang83) [reply twister2/task/src/main/java/edu/iu/dsc/tws/task/api/schedule/TaskInstanceId.java]: Removed those comments

File: twister2/task/src/main/java/edu/iu/dsc/tws/task/graph/DataFlowTaskGraph.java (hunk 4 above)
Comment (kannang83) [reply twister2/task/src/main/java/edu/iu/dsc/tws/task/graph/DataFlowTaskGraph.java]: Remove that empty statement.

**Pull request merged**